# Estimation And Confidence Intervals

Data:- A total of 15 print-heads were randomly selected and tested until failure. The durability of each print-head (in millions of characters) was recorded as follows:
1.13, 1.55, 1.43, 0.92, 1.25, 1.36, 1.32, 0.85, 1.07, 1.48, 1.20, 1.33, 1.18, 1.22, 1.29


In [7]:
data = [1.13, 1.55, 1.43, 0.92, 1.25, 1.36, 1.32, 0.85, 1.07, 1.48, 1.20, 1.33, 1.18, 1.22, 1.29]


# A) Build 99% Confidence Interval Using Sample Standard Deviation

Assuming the sample is representative of the population, construct a 99% confidence interval for the mean number of characters printed before the print-head fails using the sample standard deviation. Explain the steps you take and the rationale behind using the t-distribution for this task.


In [8]:
import numpy as np
import scipy.stats as stats

#given sample data
data = [1.13, 1.55, 1.43, 0.92, 1.25, 1.36, 1.32, 0.85, 1.07, 1.48, 1.20, 1.33, 1.18, 1.22, 1.29]
n = len(data)
mean = np.mean(data)
std_dev = np.std(data, ddof=1)  # Sample SD

# t-critical value
t_crit = stats.t.ppf(0.995, df=n-1)

# ME
margin_of_error = t_crit * (std_dev / np.sqrt(n))

# Ci
ci_lower = mean - margin_of_error
ci_upper = mean + margin_of_error



print(f"Lolets wer extrem {ci_lower}\n Upper extreme {ci_upper}")



Lolets wer extrem 1.0901973384384906
 Upper extreme 1.3871359948948425


# Explanation of Steps and Why I Used the t-distribution

1: Calculated the basics
I started by finding the mean and sample standard deviation of the given data (15 print-heads). Since this was a destructive test, the sample size was small, just 15 values.

2: Used the t-distribution
I used the t-distribution instead of the normal (z) distribution because:

The sample size is small (less than 30),

we don’t know the population standard deviation, only the sample one.

This is the standard rule — if Populqtion SD is unknown and the sample is small, we use t-distribution to be on the safer side. It has wider tails, which just means it gives a bigger margin for error when we’re less certain.

3: Found the t-critical value
Using Python’s scipy.stats.t.ppf function (or I could have used a table), I got the t-critical value for a 99% confidence level and 14 degrees of freedom (n - 1).

4: Calculated the margin of error
Then, I plugged everything into the margin of error formula

5:Built the confidence interval
 Lower extrem 1.0901973384384906
 Upper extreme 1.3871359948948425


# B) 99% Confidence Interval Using Known Population Standard Deviation

If it were known that the population standard deviation is 0.2 million characters, construct a 99% confidence interval for the mean number of characters printed before failure.



In [9]:

sigma = 0.2

z_crit = stats.norm.ppf(0.995)

# ME
margin_of_error_known_sigma = z_crit * (sigma / np.sqrt(n))

# CI
ci_lower_known = mean - margin_of_error_known_sigma
ci_upper_known = mean + margin_of_error_known_sigma

print(f"Lower Known {ci_lower_known}\n Upper known {ci_upper_known}")


Lower Known 1.1056514133957607
 Upper known 1.3716819199375725


# Explanation steps

1: Used the same sample mean
I already had the sample mean from earlier, so I reused that. The data and sample size remained the same, so nothing changed here.

2: Population standard deviation is known
This time, I was told that the population standard deviation is 0.2. That makes a big difference because when σ is known, we don’t need to estimate it from the sample. This allows us to use the z-distribution.

 3: Chose the z-distribution
Since σ is known, I used the z-distribution instead of the t-distribution. Even though the sample size is still small (n = 15), the key thing is that we now have the actual population standard deviation. That’s why the z-distribution is the correct choice.

 4: Found the z-critical value
For a 99% confidence interval, I needed the z-critical value that corresponds to 0.5% in each tail. The value comes out to approximately 2.576, which I used to calculate the margin of error.

5: Calculated the margin of error
Using the known population standard deviation, I calculated the margin of error. This gives the range above and below the sample mean that the true mean is likely to fall into.
 6: Constructed the confidence interval
Finally, I added and subtracted the margin of error from the sample mean to get the lower and upper bounds of the 99% confidence interval.

Why z-distribution was used
I used the z-distribution here because the population standard deviation was known. That makes the estimate of the mean more reliable, so the z-distribution (which has thinner tails than the t-distribution) is the proper choice.
Also, using z results in a slightly narrower interval, since there’s less uncertainty when we know σ.
